File: C++/diagonal_negativos.cpp
#include <iostream>

using namespace std;

int main(){

    int n, negativos = 0;
    cout << "Qual a ordem da matriz? ";
    cin >> n;

    int mat[n][n];

    for (int i = 0; i < n; i++){
        for (int j = 0; j < n; j++){
            cout << "Elemento [" << i << "," << j << "]: ";
            cin >> mat[i][j];
        }
    }

    cout << "DIAGONAL PRINCIPAL:" << endl;
    for (int i = 0; i < n; i++){
        cout << mat[i][i] << " ";
    }

    for (int i = 0; i < n; i++){
        for (int j = 0; j < n; j++){
            if (mat[i][j] < 0){
                negativos++;
            }
        }
    }

    cout << endl << "QUANTIDADE DE NEGATIVOS = " << negativos << endl;

    return 0;
}

File: Java/diagonal_negativos.java
import java.util.Locale;
import java.util.Scanner;

public class diagonal_negativos {

    public static void main(String[] args) {

        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

       System.out.print("Qual a ordem da matriz? ");
       int n = sc.nextInt();

       int[][] mat = new int [n][n];

       for (int i = 0; i < n; i++){
           for (int j = 0; j < n; j++){
               System.out.print("Elemento [" + i + "," + j + "]: ");
               mat[i][j] = sc.nextInt();
           }
       }

       System.out.println("DIAGONAL PRINCIPAL: ");
       for (int i = 0; i < n; i++){
           System.out.print(mat[i][i] + " ");
       }
       System.out.println();

       int negativos = 0;
        for (int i = 0; i < n; i++){
            for (int j = 0; j < n; j++){
                if (mat[i][j] < 0){
                    negativos += 1;
                }
            }
        }

        System.out.println("QUANTIDADE DE NEGATIVOS = " +  negativos);

        sc.close();

    }
}
